samples/net/lwm2m_client/src/lwm2m-client.c

```diff
@@ -265,6 +277,65 @@ static void observe_cb(enum lwm2m_observe_event event,
 	}
 }
 
+static void on_net_event_l4_disconnected(void)
+{
+	LOG_INF("Disconnected from network");
+	lwm2m_engine_pause();
+}
+
+static void on_net_event_l4_connected(void)
+{
+	LOG_INF("Connected to network");
+	k_sem_give(&network_connected_sem);
+	lwm2m_engine_resume();
+}
+
+static void l4_event_handler(struct net_mgmt_event_callback *cb,
+			     uint32_t event,
+			     struct net_if *iface)
+{
+	switch (event) {
+	case NET_EVENT_L4_CONNECTED:
+		LOG_INF("IP Up");
+		on_net_event_l4_connected();
+		break;
+	case NET_EVENT_L4_DISCONNECTED:
+		LOG_INF("IP down");
+		on_net_event_l4_disconnected();
+		break;
+	default:
+		break;
+	}
+}
+
+static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
+				       uint32_t event,
+				       struct net_if *iface)
+{
+	if (event == NET_EVENT_CONN_IF_FATAL_ERROR) {
+		LOG_ERR("Fatal error received from the connectivity layer");
+		return;
+	}
+}
+
+static int connect(void)
+{
+	int ret;
+
+	LOG_INF("Connecting to network");
+
+	ret = conn_mgr_if_connect(net_if_get_default());
+	if (ret) {
```

Comment:
What if we deal with an interface, that does not require connectivity? There will be no binding defined, and the function will fail. IMO, if we want to support manual connection in the sample, errors should be silently ignored.